common/pkg/capi/clustertopology/handlers/mutation/meta.go

```diff
@@ -5,22 +5,44 @@ package mutation
 
 import (
 	"context"
-	"strings"
 
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	bootstrapv1 "sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1beta1"
+	controlplanev1 "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1beta1"
 	runtimehooksv1 "sigs.k8s.io/cluster-api/exp/runtime/hooks/api/v1alpha1"
+	"sigs.k8s.io/cluster-api/exp/runtime/topologymutation"
 
 	"github.com/d2iq-labs/capi-runtime-extensions/common/pkg/capi/clustertopology/handlers"
 )
 
+type MetaMutater interface {
```

Comment:
@mhrabovcin Need to figure out how to massage http proxy handler to implement this interface. Problem is the cluster retrieval needs access to the request. Any ideas?

Reply:
OK figured it out, set cluster key in context